share/commands/server-upgrade-techniques

```diff
@@ -24,39 +25,157 @@ INITIAL=false
 OVERRIDE=false
 FORCE=false
 PARENT_COMMIT=""
+SET_OPTION=false
+KEY_OPTION=""
+VALUE_OPTION=""
+CONFIGURATION_FILE=/opt/rudder/etc/rudder.conf
 
-while getopts "uiofc:" opt; do
-  case $opt in
-    u)
+# Output usage
+function usage()
+{
+  echo "rudder server upgrade-techniques"
+  echo "upgrade techniques in the configuration repository from the packaged ones"
+  echo "This command will replace the techniques in /var/rudder/configuration-repository/techniques"
+  echo "by the techniques found in /opt/rudder/share/techniques which is installed by rudder-technique package."
+  echo "The upgrade can take care of user defined changes."
+  echo "This command creates an update branch rudder_update with the content of current techniques first time is it run"
+  echo ""
+  echo "It can also control behaviour of automatic Techniques upgrades in Rudder 5.0 and later"
+  echo "By default, instance upgraded from Rudder 4 to 5 will not get automatic technique upgrade,"
+  echo "unless option autoupdate-technique-library as been set to true"
+
+  echo "Options:"
+  echo "--upgrade -u         : merge updated techniques into the configuration repository"
+  echo "--initial -i         : create the initial version of the update branch"
+  echo "--override -o        : override existing technique without looking for local changes"
+  echo "--force -f           : suppress any warning and run without prompting for input"
+  echo "--commit -c COMMITID : use the given commit id as the update branch origin"
+  echo "--autoupdate-technique-library -a true/false: set the auto update technique library at upgrade to true or false (for Rudder 5+)"
+  echo "--show -s: Display the currently defined options"
+}
+
+# Check if the configuration file exists, and if not, create it with default content
+function check_configuration_file_exists()
+{
+  if [ ! -f ${CONFIGURATION_FILE} ]; then
+    cat <<EOF > ${CONFIGURATION_FILE}
+[server]
+# Configure the behavior for technique update (in Rudder 5+)
+# If set to true, Rudder upgrade will automatically replace active Techniques by those installed by rudder-techniques package
+autoupdate-technique-library=false
+EOF
+  fi
+}
+
+
+# Show all defined options in the configuration file
+function show_defined_option()
+{
+  if [ ! -f ${CONFIGURATION_FILE} ]
+  then
+    echo "Configuration file does not exists, no options set"
+  else
+    echo "Defined options are"
+    grep --color=never = ${CONFIGURATION_FILE} 
+  fi
+}
+
+# Defines available options
+OPTIONS=uiofc:a:s
+LONGOPTS=upgrade,initial,override,force,commit:,autoupdate-technique-library:,show
+
+# Use pipe to be able to get exit code, without messages
```

Comment:
The explanation is : 
- Use ! to avoid failing with set -e
- Use $PIPESTATUS to get the right return code even with !